tests/integration/test_logic.py

```diff
@@ -80,18 +148,112 @@ def test_isclose():
     out_num = num.isclose(in1_num, in2_num)
     assert np.array_equal(out_np, out_num)
 
+
+SHAPES_BROADCASTING = (
+    (1, 3),
+    (2, 3),
+    (1, 2, 3),
+    (2, 2, 3),
+    (1,),
+    (1, 1),
+    (1, 2, 1),
+    (2, 2, 1),
+)
+
+
+@pytest.mark.parametrize(
+    "shape_b", SHAPES_BROADCASTING, ids=lambda shape_b: f"(shape_b={shape_b})"
+)
+def test_isclose_broadcast(shape_b):
+    len_in_arr = 20
+    in1_np = np.random.rand(len_in_arr)
+    in2_np = in1_np + np.random.uniform(low=5e-09, high=2e-08, size=len_in_arr)
+
+    shape_a = (3,)
+    size_a = np.prod(shape_a)
+    a_np = np.array([in1_np[i % len_in_arr] for i in range(size_a)]).reshape(
+        shape_a
+    )
+
+    size_b = np.prod(shape_b)
+    b_np = np.array([in2_np[i % len_in_arr] for i in range(size_b)]).reshape(
+        shape_b
+    )
+
+    a_num = num.array(a_np)
+    b_num = num.array(b_np)
+    out_np = np.isclose(a_np, b_np)
+    out_num = num.isclose(a_num, b_num)
+    assert np.array_equal(out_np, out_num)
+
+
+EMPTY_ARRAY_PAIRS = (
+    ([], []),
+    ([], [[]]),
+    ([[]], [[]]),
+)
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    EMPTY_ARRAY_PAIRS,
+)
+def test_isclose_empty_arrays(a, b):
+    out_np = np.isclose(a, b)
+    out_num = num.isclose(a, b)
+    assert np.array_equal(out_np, out_num)
+
+
+@pytest.mark.parametrize(
+    ("rtol", "atol"),
+    ((1e-04, 1e-06), (1e-06, 1e-09)),
+)
+def test_isclose_arrays_rtol_atol(rtol, atol):
+    in1_np = np.random.rand(10)
+    in2_np = in1_np + np.random.uniform(low=5e-09, high=2e-08, size=10)
+    in1_num = num.array(in1_np)
+    in2_num = num.array(in2_np)
+
+    out_np = np.isclose(in1_np, in2_np, rtol=rtol, atol=atol)
+    out_num = num.isclose(in1_num, in2_num, rtol=rtol, atol=atol)
+    assert np.array_equal(out_np, out_num)
+
+
+@pytest.mark.parametrize(
+    "equal_nan", (False, pytest.param(True, marks=pytest.mark.xfail))
+)
+def test_isclose_euqal_nan(equal_nan):
+    # If equal_nan is True,
+    # In Numpy, it pass
+    # In cuNumeric, it raises NotImplementedError
     weird_values = [np.inf, -np.inf, np.nan, 0.0, -0.0]
     weird_pairs = tuple(combinations_with_replacement(weird_values, 2))
     in1_np = np.array([x for x, _ in weird_pairs])
     in2_np = np.array([y for _, y in weird_pairs])
```

Comment:
modifier seems superfluous, suggest to keep name simpler
```suggestion
    values = [np.inf, -np.inf, np.nan, 0.0, -0.0]
    pairs = tuple(combinations_with_replacement(values, 2))
    in1_np = np.array([x for x, _ in pairs])
    in2_np = np.array([y for _, y in pairs])
```

Reply:
Fixed. Thanks.